From 3b83cfe61c6eb457f4dfef57ef718a23af111e5a Mon Sep 17 00:00:00 2001
From: Nic Costa <nic.costa@arm.com>
Date: Thu, 22 Oct 2020 16:34:58 +0000
Subject: [PATCH 04/15] Call a restart script during fota

During a FOTA, it may not be necessary or desired to reboot the OS,
so instead edge-core calls out to a shell script to allow a
systems integrator to decide what needs to happen during a FOTA
restart, whether that's rebooting the OS or restarting only the
edge-core process.
---
 .../source/update_client_hub_state_machine.c                | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/lib/mbed-cloud-client/update-client-hub/source/update_client_hub_state_machine.c b/lib/mbed-cloud-client/update-client-hub/source/update_client_hub_state_machine.c
index b884d46..d2fb113 100644
--- a/lib/mbed-cloud-client/update-client-hub/source/update_client_hub_state_machine.c
+++ b/lib/mbed-cloud-client/update-client-hub/source/update_client_hub_state_machine.c
@@ -1253,11 +1253,7 @@ void ARM_UC_HUB_setState(arm_uc_hub_state_t new_state)
                 UC_HUB_TRACE("ARM_UC_HUB_STATE_REBOOT");
 
                 // Firmware activated, now reboot the system to apply the new image.
-#if defined(ARM_UC_PROFILE_MBED_CLIENT_LITE) && (ARM_UC_PROFILE_MBED_CLIENT_LITE == 1)
-                arm_uc_plat_reboot();
-#else
-                pal_osReboot();
-#endif
+                system("/snap/" SNAPCRAFT_PROJECT_NAME "/current/wigwag/mbed/arm_update_restart.sh");
 
                 /* Reboot not implemented on this platform.
                    Go to idle state.
-- 
2.17.1

